internals/overlord/overlord.go

```diff
@@ -118,6 +118,13 @@ func New(opts *Options) (*Overlord, error) {
 		extension: opts.Extension,
 	}
 
+	// Enable the ensure timer before overlord managers are added. This allows
+	// managers to call state.EnsureBefore() safely before the overlord loop
+	// is started, as soon as they have a valid state instance. The overlord
+	// loop will always ensure on entry, so any pending ensure requests are
+	// serviced as soon as the overlord loop is started.
+	o.ensureTimerSetup()
```

Comment:
if that's the goal personally I would have just added a bool flag ensureTimerInitialized that is to be used under the lock, and set it to true in ensureTimerSetup and simply taught EnsuseBefore to be a noop if it's false instead of the current panic

Reply:
strickly speaking you could even just use the nil check instead of a flag. I also wonder whether we need to worry though about what should happen during stopping of the Overlord in a world where EnsureBefore can be called anywhere

Reply:
OK makes a lot of sense ... super!

Reply:
@pedronis I think your proposal makes a lot of sense, it completely avoids having to touch functional timer code, which as you say is likely a good thing due to the known implementation issues before Go 1.23. Please see the new proposal.